frontend/src/components/personal/PreferencesPage.tsx

```diff
@@ -9,13 +9,22 @@ import {useStyles} from './ProfilePreferencesPage.styles'
 import {RootState} from '../../reducers/index'
 import './PreferencesPage.scss'
 import { userLogOut } from '../../actions/authActions'
+import { timeDiffChecker } from '../../api/TokenRefreshInstance'
+import { TWENTYTHOUSAND } from '../../constants/valuableNumbers'
 
 export default function PreferencesPage (): React.ReactElement {
   const dispatch = useDispatch()
   
+  
+  useEffect(() => {
+    dispatch(getMyData())
+  }, [dispatch])  
+  
   useEffect(() => {
-        dispatch(getMyData())
-    }, [dispatch])  
+    setInterval(() => {
+     timeDiffChecker()
+    }, TWENTYTHOUSAND)
```

Comment:
give a better name for `TWENTYTHOUSAND`
Also, can we just use `setTimeout` to call a refresh once it exact time?

Reply:
setTimeout is called once, while we need check the time expiration regularly

Reply:
So, as we know when the token is expiring, why just not use `setTimeout` with concrete time? We know current time, we know expiration time. Subtract one from another and small ~10 seconds period to to refresh just before expiration, and use it in `setTimeout`.